### PR TITLE
[chore] Fix more load test pipeline issues

### PR DESCRIPTION
**Pull request by mowies**
**Description:** <Describe what has changed.>

This PR fixes more issues with the load test pipeline. Specifically, it adjusts some things around artifact up/download which is needed as a result of the breaking changes in GitHub Actions artifacts listed [here](https://github.com/actions/upload-artifact#breaking-changes:~:text=Uploading%20to%20the%20same%20named,you%20will%20encounter%20an%20error.).
Also, it removes the issue generation step which seems far outdated as it's failing with missing CircleCI env variables.

Also, I noticed that a few of the load tests fail consistently. To test. the pipeline, I had to comment them out. Not sure, what to do with them...

**Link to tracking Issue:**
Part of https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33577
Follow up to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33860

**Testing:** Testing done on my personal fork. Link to working pipeline run:
[Green run with some broken tests commented out, but pipeline working](https://github.com/mowies/opentelemetry-collector-contrib/actions/runs/9789652148)
[Red run with broken tests](https://github.com/mowies/opentelemetry-collector-contrib/actions/runs/9790395589)

**Documentation:** none added

### CODE REVIEW

File: .github/workflows/load-tests.yml [hunk 1]
```diff
@@ -96,6 +96,7 @@ jobs:
         env:
           TEST_ARGS: "-test.run=${{ matrix.test }}"
       - name: Set results filename
+        if: ${{ failure() || success() }}
```
Comment (mx-psi) [inline .github/workflows/load-tests.yml:99]: I don't know much about the fine details of this vs `always()`. Do you think we could use `always()` here? (https://docs.github.com/en/actions/learn-github-actions/expressions#example-of-always)

Comment (mowies) [reply .github/workflows/load-tests.yml:99]: i thought about that as well and opted for using the same expression that is used in other places.
The main difference is that `always()` also runs when a job is cancelled which is probably not needed here.

**Comment by mx-psi** [comment]
Can we add an issue to track adding

> Also, it removes the issue generation step which seems far outdated as it's failing with missing CircleCI env variables.


back?

File: .github/workflows/load-tests.yml (hunk 1 above)
Comment (mx-psi) [reply .github/workflows/load-tests.yml:99]: Makes sense to be consistent, thanks for clarifying

**Comment by mowies** [comment]
@mx-psi I created https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33910 to track it

**Review by mx-psi — APPROVED** [review]

**Pull request merged**